webui/src/lib/components/repository/changes.jsx

```diff
@@ -69,7 +69,7 @@ export const TreeItemRow = ({ entry, repo, reference, leftDiffRefID, rightDiffRe
     if (itemType.response === TreeItemType.Object) {
         return <>
             <ObjectTreeEntryRow key={entry.path + "entry-row"} entry={entry} relativeTo={relativeTo}
-                                depth={depth === 0 ? 0 : depth + 1} onRevert={onRevert} repo={repo}
+                                depth={depth === 0 ? 0 : depth + 1} onRevert={!repo?.read_only ? onRevert : null} repo={repo}
```

Comment:
Shouldn't we disable this button?  In which case I say: if the action comes through here, let it run through to the backend and get an error.  GUI will be less confusing because the user will get _some_ accurate readable indication, it will simply be ugly.

Reply:
Fixed